core/servers/content/web_handlers/webfinger.js

```diff
@@ -0,0 +1,135 @@
+const { ServerModule } = require('../../../server_module');
+const Config = require('../../../config').get;
+const { Errors } = require('../../../enig_error');
+
+const WebServerPackageName = require('../web').moduleInfo.packageName;
+
+const _ = require('lodash');
+const User = require('../../../user');
+const { result } = require('lodash');
+const Log = require('../../../logger').log;
+
+exports.moduleInfo = {
+    name: 'WebFinger',
+    desc: 'A simple WebFinger Handler',
+    author: 'NuSkooler',
+    packageName: 'codes.l33t.enigma.web.handler.finger',
+};
+
+exports.getModule = class WebFingerServerModule extends ServerModule {
+    constructor() {
+        super();
+    }
+
+    init(cb) {
+        if (!_.get(Config(), 'contentServers.web.handlers.webFinger.enabled')) {
+            return cb(null);
+        }
+
+        const { getServer } = require('../../../listening_server');
+
+        // we rely on the web server
+        this.webServer = getServer(WebServerPackageName);
+        if (!this.webServer || !this.webServer.instance.isEnabled()) {
+            return cb(Errors.UnexpectedState('Cannot access web server!'));
+        }
+
+        this.webServer.instance.addRoute({
+            method: 'GET',
+            path: /^\/\.well-known\/webfinger\/?\?/,
+            handler: this._webFingerRequestHandler.bind(this),
+        });
+
+        return cb(null);
+    }
+
+    _webFingerRequestHandler(req, resp) {
+        const url = new URL(req.url, `https://${req.headers.host}`);
+
+        const resource = url.searchParams.get('resource');
+        if (!resource) {
+            return this.webServer.instance.respondWithError(
+                resp,
+                400,
+                '"resource" is required',
+                'Missing "resource"'
+            );
+        }
+
+        this._getUser(resource, resp, (err, user, accountName) => {
+            if (err) {
+                // |resp| already written to
+                return Log.warn({ error: err.message }, `WebFinger failed: ${req.url}`);
+            }
+
+            const body = JSON.stringify({
+                subject: `acct:${accountName}`,
+                links: [this._profilePageLink(user)],
+            });
+
+            const headers = {
+                'Content-Type': 'application/jrd+json',
+                'Content-Length': body.length,
+            };
+
+            resp.writeHead(200, headers);
+            return resp.end(body);
+        });
+    }
+
+    _profilePageLink(user) {
+        const href = this.webServer.instance.buildUrl(`/wf/@${user.username}`);
+        return {
+            rel: 'http://webfinger.net/rel/profile-page',
+            type: 'text/plain',
```

Comment:
My thinking is a text/plain or perhaps a HTML simple profile page are best to return. The spec allows for whatever we like, but e.g. Mastodon may want specifics.